src/splitEvery.js

```diff
@@ -0,0 +1,33 @@
+var _curry2 = require('./internal/_curry2');
+var slice = require('./slice');
+
+
+/**
+ * Splits a collection into slices of the specified length.
+ *
+ * Dispatches to its second argument's `slice` method if present. As a
+ * result, one may replace `[a]` with `String` in the type signature.
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig Number -> [a] -> [[a]]
+ * @param {Number} n
+ * @param {Array} list
+ * @return {Array}
+ * @example
+ *
+ *      R.splitEvery(3, [1, 2, 3, 4, 5, 6, 7]); //=> [[1, 2, 3], [4, 5, 6], [7]]
+ *      R.splitEvery(3, 'foobarbaz'); //=> ['foo', 'bar', 'baz']
+ */
+module.exports = _curry2(function splitEvery(n, list) {
+  if (n <= 0) {
+    throw new Error('First argument to splitEvery must be a positive integer');
```

Comment:
this is new. The only other functions where we deliberately throw are 

* `concat` because we can't figure out how to concatenate the arguments
* `constructN` because the arity `N > 10` 
* `nAry` 'First argument to nAry must be a non-negative integer no greater than ten'

I understand the rationale for throwing here; we really don't know what it means to split after every -2 elements. But i'm sure there are lots of other places in the code we are not doing such checking. i believe highland returns the whole stream/list in this situation, which may be reasonable; but may lead to difficult to trace bugs.